qrmi/dependencies/pasqal_cloud_client/src/client.rs

```diff
@@ -0,0 +1,263 @@
+pub struct CreateBatchResponseData {
+    pub id: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct GetBatchResponseData {
+    pub status: BatchStatus,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct CancelBatchResponseData {}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct GetBatchResultLinksResponseData {
+    pub results_links: HashMap<String, String>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct Job {
+    pub runs: i32,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct Batch {
+    pub sequence_builder: String,
+    pub jobs: Vec<Job>,
+    pub device_type: DeviceType,
+    pub project_id: String,
+}
+
+impl Client {
+    pub async fn get_device(
+        &self,
+        device_type: DeviceType,
+    ) -> Result<Response<GetDeviceResponseData>> {
+        let url = format!(
+            "{}/core-fast/api/v1/devices?device_type={}",
+            self.base_url, device_type,
+        );
+        self.get(&url).await
+    }
+    /// Pasqal Cloud works with batches of jobs rather than
+    /// individual jobs, see:
+    /// https://docs.pasqal.com/cloud/batches/
+    pub async fn create_batch(
+        &self,
+        sequence: String,
+        job_runs: i32,
+        device_type: DeviceType,
+    ) -> Result<Response<CreateBatchResponseData>> {
+        let url = format!("{}/core-fast/api/v1/batches", self.base_url);
+        let batch = Batch {
+            sequence_builder: sequence,
+            jobs: Vec::from([Job { runs: job_runs }]),
+            device_type,
+            project_id: self.project_id.clone(),
+        };
+        self.post(&url, batch).await
+    }
+
+    pub async fn cancel_batch(&self, batch_id: &str) -> Result<Response<CancelBatchResponseData>> {
+        let url = format!(
+            "{}/core-fast/api/v2/batches/{}/cancel",
+            self.base_url, batch_id
+        );
+        self.patch(&url).await
+    }
+
+    pub async fn get_batch(&self, batch_id: &str) -> Result<Response<GetBatchResponseData>> {
+        let url = format!("{}/core-fast/api/v2/batches/{}", self.base_url, batch_id);
+        self.get(&url).await
+    }
+
+    pub async fn get_batch_results(&self, batch_id: &str) -> Result<String> {
+        match self.get_batch_result_links(batch_id).await {
+            Ok(resp) => {
+                let results_links = resp.data.results_links;
+                // by design only one job
+                let mut results: String = "".to_string();
+                for (i, (_job_id, result_link)) in results_links.iter().enumerate() {
+                    if i > 0 {
+                        bail!(format!(
+                            "Unexpected multiple jobs in one Pasqal cloud batch"
+                        ));
+                    };
+                    results = reqwest::get(result_link).await?.text().await?;
+                }
+                if results == *"" {
+                    bail!(format!("No results found"));
+                }
+                Ok(results)
```

Comment:
> this is ugly, at least we should return the hashmap with <job_id, results> and pick the only one element on the qrmi main module.

From Aureliens first implementation, still todo. but I want to finish the Spank side first